.github/workflows/pr-build-image.yaml

```diff
@@ -52,69 +57,37 @@ jobs:
           git fetch base-origin ${{ github.event.pull_request.base.ref }}
           git merge --no-edit base-origin/${{ github.event.pull_request.base.ref }}
 
-      - name: Get the last commit short SHA of the PR
-        uses: ./.github/actions/get-sha
-
-      - name: Check if Image Already Exists
-        id: image-check
-        run: |
-          IMAGE_TAG_COMMIT="pr-${{ github.event.number }}-${{ env.SHORT_SHA }}"
-          IMAGE_NAME_COMMIT="${{ env.REGISTRY }}/rhdh-community/rhdh:${IMAGE_TAG_COMMIT}"
-
-          # Check if any image exists for the specific commit
-          IMAGE_EXISTS_COMMIT=$(curl -s "https://quay.io/api/v1/repository/rhdh-community/rhdh/tag/" | jq -r --arg tag "$IMAGE_TAG_COMMIT" '.tags[] | select(.name == $tag) | .name')
-
-          if [ -n "$IMAGE_EXISTS_COMMIT" ]; then
-            echo "Image $IMAGE_NAME_COMMIT already exists for the current commit."
-            echo "image_exists=true" >> $GITHUB_ENV
-          else
-            echo "Image $IMAGE_NAME_COMMIT does not exist for the current commit."
-            echo "image_exists=false" >> $GITHUB_ENV
-          fi
-
-      - name: Determine Changed Files
-        id: changes
-        run: |
-          BASE_COMMIT=${{ github.event.pull_request.base.sha }}
-          HEAD_COMMIT=${{ github.event.pull_request.head.sha }}
-
-          CHANGED_FILES=$(git diff --name-only "$BASE_COMMIT" "$HEAD_COMMIT")
-
-          echo "Changed files:"
-          echo "$CHANGED_FILES"
-
-          # Check if changes are only in .ibm/ or e2e-tests/
-          if echo "$CHANGED_FILES" | grep -qvE '^(e2e-tests/|\.ibm/)'; then
-            echo "Changes detected outside the e2e-tests or .ibm folders. Build required."
-            echo "relevant_changes=true" >> $GITHUB_ENV
-          else
-            echo "No relevant changes detected."
-            echo "relevant_changes=false" >> $GITHUB_ENV
-          fi
-
       - name: Build and Push with Buildx
-        if: env.relevant_changes == 'true' && env.image_exists == 'false'
         uses: ./.github/actions/docker-build
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_TOKEN }}
           imageName: rhdh-community/rhdh
           imageTags: |
-            type=ref,prefix=pr-,suffix=-${{ env.SHORT_SHA }},event=pr
+            type=ref,prefix=pr-,suffix=-${{ needs.check-image.outputs.short_sha }},event=pr
```

Comment:
since the check-image stage sets `${{ env.SHORT_SHA }}` why not just use that env?